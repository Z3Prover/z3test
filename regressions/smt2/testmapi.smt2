(push)
(declare-fun f (Int Int) Bool)
(declare-const il (Seq Int))
(assert (= (seq.mapi f 0 il) (seq.++ (seq.unit false) (seq.unit true) (seq.unit false))))
(check-sat)
(pop)

(push)
(declare-const il (Seq Int))
(declare-const F (Array Int Int Bool))
(assert (= (seq.mapi F 0 il) (seq.++ (seq.unit false) (seq.unit true) (seq.unit false))))
(check-sat)
(pop)

(push)
(declare-const f (Array Int Int Bool))
(declare-const il_1 (Seq Int))
(declare-const il_2 (Seq Int))
(simplify (seq.mapi f 1 (as seq.empty (Seq Int))))
(simplify (seq.mapi f 2 (seq.++ il_1 il_2)))
(simplify (seq.mapi f 3 (seq.unit 4)))
(pop)

(push)
(declare-fun f (Int Int) Bool)
(declare-const il (Seq Int))
(declare-const bl (Seq Bool))
(assert (= (seq.mapi f 1 il) bl))
(assert (> (seq.len il) 1))
(check-sat)
(pop)

