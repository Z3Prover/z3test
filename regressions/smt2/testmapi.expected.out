sat
sat
(as seq.empty Seq)
(seq.++ (seq.mapi f 2 il_1) (seq.mapi f (+ 2 (seq.len il_1)) il_2))
(seq.unit (select f 3 4))
sat
